h2o-r/h2o-package/R/admissibleml.R

```diff
@@ -0,0 +1,150 @@
+#' Calculate intersectional fairness metrics.
+#'
+#' @param model H2O Model
+#' @param frame Frame used to calculate the metrics.
+#' @param protected_cols List of categorical columns that contain sensitive information
+#'                          such as race, gender, age etc.
+#' @param reference List of values corresponding to a reference for each protected columns.
+#'        If set to NULL, it will use the biggest group as the reference.
+#' @param favorable_class Positive/favorable outcome class of the response.
+#'
+#' @return Dictionary of frames. One frame is the overview, other frames contain dependence
+#'         of performance on threshold for each protected group.
+#' @export
+h2o.calculate_fairness_metrics <- function(model, frame, protected_cols, reference, favorable_class) {
+  model_id <- if (is.character(model)) model else model@model_id
+  if (is.null(h2o.keyof(frame)))
+    head(frame, n = 1) # force evaluation of frame (in case it was manipulated before (e.g. subset))
+  list_to_string <- function(entries) paste0("[\"", paste0(entries, collapse = "\", \"") ,"\"]")
+  expr <- sprintf("(fairnessMetrics %s %s %s %s \"%s\")",
+                  model_id ,
+                  h2o.keyof(frame),
+                  list_to_string(protected_cols),
+                  list_to_string(reference),
+                  favorable_class)
+  lst <- h2o.rapids(expr)
+  res <- list()
+  for (i in seq_along(lst$map_keys$string)) {
+    res[[lst$map_keys$string[[i]]]] <- as.data.frame(h2o.getFrame(lst$frames[[i]]$key$name))
+  }
+  res
+}
+
+
+.get_corrected_variance <- function(fm) {
+    # From De-biasing bias measurement https://arxiv.org/pdf/2205.05770.pdf
+    max(0, var(fm[["accuracy"]], na.rm = TRUE) - mean((fm[["accuracy"]] * (1 - fm[["accuracy"]])) /
+                                                        fm[["total"]], na.rm = TRUE))
+}
+
+
+#' Create a frame containing aggregations of intersectional fairness across the models.
+#'
+#' @param models List of H2O Models
+#' @param newdata H2OFrame
+#' @param protected_cols List of categorical columns that contain sensitive information such as race, gender, age etc.
+#' @param reference List of values corresponding to a reference for each protected columns.
+#'                  If set to NULL, it will use the biggest group as the reference.
+#' @param favorable_class Positive/favorable outcome class of the response.
+#' @return frame containing aggregations of intersectional fairness across the models
```

Comment:
Can we add an example in the docstring?  I realize it will be a bit long... but maybe it would be helpful?